lib/sprockets/less/v2/less_template.rb

```diff
@@ -0,0 +1,252 @@
+          @filename = options[:filename]
+          @source = options[:data]
+          @options = options.merge(@default_options)
+          @importer_class = options[:importer]
+          @less_config = options[:less_config] || {}
+          @input = options
+          @functions = Module.new do
+            include Sprockets::Helpers if defined?(Sprockets::Helpers)
+            include Sprockets::Less::Utils.get_class_by_version('Functions')
+            include options[:functions] if options[:functions]
+            class_eval(&block) if block_given?
+          end
+        end
+
+        @less_functions_initialized = false
+        class << self
+          attr_accessor :less_functions_initialized
+          alias less_functions_initialized? less_functions_initialized
+          # Templates are initialized once the functions are added.
+          def engine_initialized?
+            less_functions_initialized?
+          end
+        end
+
+        # Add the Less functions if they haven't already been added.
+        def initialize_engine
+          return if self.class.engine_initialized?
+
+          if Sprockets::Less.add_less_functions
+            begin
+              require 'sprockets/helpers'
+              require 'sprockets/less/functions'
+              self.class.less_functions_initialized = true
+            rescue LoadError; end
+          end
+        end
+
+        def call(input)
+          @input = input
+          @filename = input[:filename]
+          @source   = input[:data]
+          @context  = input[:environment].context_class.new(input)
+          run
+        end
+
+        def render(context, _empty_hash_wtf)
+          @context = context
+          run
+        end
+
+
+        def less_engine(data, less_options, css_options)
+          ::Less.Parser['sprockets_context'] = context
+          parser = ::Less::Parser.new(less_options)
+          parser.tree.extend_js @functions
+          engine = parser.parse(data)
+          engine.to_css(css_options)
+        end
+
+        def run
+          css = retrieve_from_cache_store
+          if css.nil?
+            data = Sprockets::Less::Utils.read_file_binary(filename, options)
+            new_data, dependencies = process_dependencies(data)
+
+            css  = less_engine(new_data, less_options, css_options)
+            store_into_cache_store(css)
+
+            less_dependencies = Set.new([filename])
+            if context.respond_to?(:metadata)
+              dependencies.map do |dependency|
+                less_dependencies << dependency.to_s
+                context.metadata[:dependencies] << Sprockets::URIUtils.build_file_digest_uri(dependency.to_s)
+              end
+              context.metadata.merge(data: css, less_dependencies: less_dependencies)
+            else
+              css
+            end
+          else
+            css
+          end
```

Comment:
Line 135 exit point is easily overlooked due to too much nesting. Turn `if css.nil?` and `if context.respond_to?(:metadata)` into guard clauses that `return css`.